src/lambda/handler-runner/go-runner/GoRunner.js

```diff
@@ -0,0 +1,186 @@
+import { EOL } from 'os'
+import { writeFileSync, readFileSync, mkdirSync, rmdirSync } from 'fs'
+import { delimiter, resolve, parse as pathParse } from 'path'
+import { spawnSync } from 'child_process'
+import AWS from 'aws-sdk'
+
+const { parse, stringify } = JSON
+const { cwd } = process
+
+const MIN_ALLOWED_SESSION_DURATION_S = 900
+const SESSION_TIMEOUT_MS = 850 * 1000
+const PAYLOAD_IDENTIFIER = 'offline_payload'
+
+export default class GoRunner {
+  #env = null
+  #handlerPath = null
+  #tmpPath = null
+  #tmpFile = null
+  #profile = null
+  #handle = null
+  #credentials = null
+  #goEnv = null
+
+  constructor(funOptions, env, v3Utils) {
+    const { handlerPath, provider } = funOptions
+
+    this.#env = env
+    this.#handlerPath = handlerPath
+    this.#profile = provider.profile || 'default'
+
+    if (v3Utils) {
+      this.log = v3Utils.log
+      this.progress = v3Utils.progress
+      this.writeText = v3Utils.writeText
+      this.v3Utils = v3Utils
+    }
+
+    // Make sure we have the mock-lambda runner
+    spawnSync('go', ['get', 'github.com/icarus-sullivan/mock-lambda'])
+  }
+
+  cleanup() {
+    // Clean up session timeout
+    if (this.#handle !== null) {
+      clearTimeout(this.#handle)
+    }
+
+    try {
+      rmdirSync(this.#tmpPath, { recursive: true })
+    } catch (e) {
+      // @ignore
+    }
+
+    this.#tmpFile = null
+    this.#tmpPath = null
+  }
+
+  _parsePayload(value) {
+    const log = []
+    let payload
+
+    for (const item of value.split(EOL)) {
+      if (item.indexOf(PAYLOAD_IDENTIFIER) === -1) {
+        log.push(item)
+      } else if (item.indexOf(PAYLOAD_IDENTIFIER) !== -1) {
+        try {
+          const {
+            offline_payload: { success, error },
+          } = parse(item)
+          if (success) {
+            payload = success
+          } else if (error) {
+            payload = error
+          }
+        } catch (err) {
+          // @ignore
+        }
+      }
+    }
+
+    // Log to console in case engineers want to see the rest of the info
+    if (this.log) {
+      this.log(log.join(EOL))
+    } else {
+      console.log(log.join(EOL))
+    }
+
+    return payload
+  }
+
+  clearCredentials() {
+    this.#credentials = null
+    this.#handle = null
+  }
+
+  async run(event, context) {
+    const { dir } = pathParse(this.#handlerPath)
+    const handlerCodeRoot = dir.split(delimiter).slice(0, -1).join(delimiter)
+    const handlerCode = readFileSync(`${this.#handlerPath}.go`, 'utf8')
+    this.#tmpPath = resolve(handlerCodeRoot, 'tmp')
+    this.#tmpFile = resolve(this.#tmpPath, 'main.go')
+
+    const out = handlerCode.replace(
+      '"github.com/aws/aws-lambda-go/lambda"',
+      'lambda "github.com/icarus-sullivan/mock-lambda"',
+    )
```

Comment:
Clone code and swap out lambda import, mock-lambda looks for the `LAMBDA_EVENT` and `LAMBDA_CONTEXT` and tries to process it.